src/gc/unix/cgroup.h

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+/*++
+
+Module Name:
+
+    cgroup.h
+
+Abstract:
+    Read memory limits for the current process
+--*/
+
+#define MAX_PATH 1024
+
+class CGroup
+{
+    char m_memory_cgroup_path[MAX_PATH];
+public:
+    CGroup();
+    bool GetPhysicalMemoryLimit(size_t* val);
+    bool GetSwapMemoryLimit(size_t* val);
+private:
+    bool FindMemoryHierarchyMount(char* mountpath);
+    bool FindCGroupPathForMemorySubsystem(char* cgrouppath);
+    bool ReadMemoryValueFromFile(const char* filename, size_t* val);
+};
+
+size_t GetRestrictedPhysicalMemoryLimit();
```

Comment:
It would look better to have these methods as static public methods on CGroup, and have all instance methods in CGroup to be private;  or move `class CGroup` to the .cpp file.